[doc] Add config of dip switches to FPGA setup (#30722)
lowRISC/opentitan: doc/getting_started/setup_fpga.md

The CW340 board should be powered via the included DC power adapter.
To this end:
1. Turn off the board by setting the *Control Power* switch (top left corner, *SW7*) to the right (towards the OpenTitan logo).
2. Ensure the *Tgt Power* switch (center of the board) is set to the right, towards the *Auto* option.
3. Plug the DC power adapter into the barrel jack (*J11*) in the top left corner of the board.
4. Use a USB-C cable to connect your PC (*host*) with the *USB-C* connector (*J28*) in the lower left corner on the board.
5. Set the jumpers *JP1* and *JP2* to select the UART0 routing:
   1. If set to FTDI the UART0 will (likely) be routed to `/dev/ttyUSB2`.
   2. If set to SAM the UART0 will be routed to `/dev/ttyACM0`.

> 6. Make sure the dip switch SW4 on the top-side of the board has all switches (1-8) in the ON position.

Reviewer: Grammar nit:

```suggestion
6. Make sure the dip switch SW4 on the top side of the board has all switches (1-8) in the ON position.
```
(or "topside", but I think I prefer "top side"). Same for "bottom-side" below.

> 6. Make sure the dip switch SW4 on the top-side of the board has all switches (1-8) in the ON position.

Reviewer: Nit - can you capitalize "dip" throughout? As it is an acronym.

```suggestion
6. Make sure the DIP switch SW4 on the top-side of the board has all switches (1-8) in the ON position.
```

7. Make sure the dip switches on the bottom-side of the board are set up as follows:
   - SW1, SW2: all switches (1 to 8) in the OFF position

>    - SW3: switch 1 in the ON position, switch 2, 3 in the OFF position.

Reviewer 2: As we discussed, check if all switches to OFF position would have any effect.

   - SW10, SW11: all switches (1 to 3) in the ON position
   - SW12: all switches (1 to 6) in the ON position
   - dip switch connected to IOB0 - IOB7: all switches (1 to 8) in the ON position

>    - dip switch connected to IOB8 - IOB12: switch 1 to 7 in the ON position, switch 8 in the OFF position.

Reviewer 2: Same here.

Author: It looks like switch 8 really matters. Turning it ON (so that all switches 1-8 are ON) causes all the tests mentioned below in this page to fail. The effect of switch 8 is captured by `dmesg` (below). Doing a diff between the logs with the switch ON and OFF I see that switch 8 in the OFF position seems to make the HyperDebug board functional/discoverable.

Interestingly, in both cases, `dmesg` captures a number of errors during initialization:

```text
device descriptor read/64, error -32
Device not responding to setup address.
device not accepting address 59, error -71
WARN: invalid context state for evaluate context command.
```

I am not sure these are documented anywhere?

Also, just out of curiosity, I tried turning switch 7 OFF (1-6 ON, 7-8 OFF). This didn't make any difference. Not to the dmesg log, nor to the tests mentioned below in this page, which all continue to pass. It could be that other tests fail, though. I guess it may make sense to have a look at the schematics rather than randomly switching things ON and OFF, but I am not sure I can make sense of it by myself :)

Here are the raw `dmesg` logs for future reference:

```text
$ dmesg -H  # Switch 1-8 turned ON
...
[  +1.265997] usb 5-1: new high-speed USB device number 39 using xhci_hcd
[  +0.125920] usb 5-1: New USB device found, idVendor=0424, idProduct=2534, bcdDevice= 1.82
[  +0.000017] usb 5-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[  +0.039666] hub 5-1:1.0: USB hub found
[  +0.000365] hub 5-1:1.0: 4 ports detected
[  +0.515071] usb 5-1.2: new high-speed USB device number 40 using xhci_hcd
[  +0.087061] usb 5-1.2: New USB device found, idVendor=0403, idProduct=6011, bcdDevice= 8.00
[  +0.000009] usb 5-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[  +0.000003] usb 5-1.2: Product: Quad RS232-HS
[  +0.000003] usb 5-1.2: Manufacturer: FTDI
[  +0.001570] ftdi_sio 5-1.2:1.0: FTDI USB Serial Device converter detected
[  +0.000059] usb 5-1.2: Detected FT4232H
[  +0.000593] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB0
[  +0.000331] ftdi_sio 5-1.2:1.1: FTDI USB Serial Device converter detected
[  +0.000029] usb 5-1.2: Detected FT4232H
[  +0.000617] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB1
[  +0.000353] ftdi_sio 5-1.2:1.2: FTDI USB Serial Device converter detected
[  +0.000027] usb 5-1.2: Detected FT4232H
[  +0.000636] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB2
[  +0.000347] ftdi_sio 5-1.2:1.3: FTDI USB Serial Device converter detected
[  +0.000048] usb 5-1.2: Detected FT4232H
[  +0.000595] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB3
[  +0.304441] usb 5-1.1: new high-speed USB device number 41 using xhci_hcd
[  +0.089516] usb 5-1.1: New USB device found, idVendor=2b3e, idProduct=c340, bcdDevice= 9.00
[  +0.000012] usb 5-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  +0.000004] usb 5-1.1: Product: ChipWhisperer CW340 - Luna Board
[  +0.000004] usb 5-1.1: Manufacturer: NewAE Technology Inc.
[  +0.000004] usb 5-1.1: SerialNumber: 51203220353053523230382037313038
[  +0.086503] cdc_acm 5-1.1:1.1: ttyACM0: USB ACM device
[  +0.000682] cdc_acm 5-1.1:1.3: ttyACM1: USB ACM device
[  +0.178289] usb 5-1.4: new full-speed USB device number 42 using xhci_hcd
[  +0.073060] usb 5-1.4: device descriptor read/64, error -32
[  +0.174978] usb 5-1.4: device descriptor read/64, error -32
[  +0.177210] usb 5-1.4: new full-speed USB device number 43 using xhci_hcd
[  +0.075144] usb 5-1.4: device descriptor read/64, error -32
[  +0.181989] usb 5-1.4: device descriptor read/64, error -32
[  +0.101145] usb 5-1-port4: attempt power cycle
[  +0.587599] usb 5-1.4: new full-speed USB device number 44 using xhci_hcd
[  +0.000154] usb 5-1.4: Device not responding to setup address.
[  +0.205180] usb 5-1.4: Device not responding to setup address.
[  +0.206935] usb 5-1.4: device not accepting address 44, error -71
[  +0.000152] usb 5-1.4: WARN: invalid context state for evaluate context command.
[  +0.072599] usb 5-1.4: new full-speed USB device number 45 using xhci_hcd
[  +0.000126] usb 5-1.4: Device not responding to setup address.
[  +0.207064] usb 5-1.4: Device not responding to setup address.
[  +0.208157] usb 5-1.4: device not accepting address 45, error -71
[  +0.000235] usb 5-1.4: WARN: invalid context state for evaluate context command.
[  +0.000113] usb 5-1-port4: unable to enumerate USB device
```

```text
$ dmesg -H  # Switch 8 turned OFF, 1-7 ON. Note the line "Product: HyperDebug CMSIS-DAP"
...
[  +2.368678] usb 5-1: new high-speed USB device number 54 using xhci_hcd
[  +0.125743] usb 5-1: New USB device found, idVendor=0424, idProduct=2534, bcdDevice= 1.82
[  +0.000016] usb 5-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[  +0.043220] hub 5-1:1.0: USB hub found
[  +0.000303] hub 5-1:1.0: 4 ports detected
[  +0.278654] usb 5-1.2: new high-speed USB device number 55 using xhci_hcd
[  +0.088250] usb 5-1.2: New USB device found, idVendor=0403, idProduct=6011, bcdDevice= 8.00
[  +0.000016] usb 5-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[  +0.000007] usb 5-1.2: Product: Quad RS232-HS
[  +0.000006] usb 5-1.2: Manufacturer: FTDI
[  +0.001679] ftdi_sio 5-1.2:1.0: FTDI USB Serial Device converter detected
[  +0.000040] usb 5-1.2: Detected FT4232H
[  +0.000608] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB0
[  +0.000331] ftdi_sio 5-1.2:1.1: FTDI USB Serial Device converter detected
[  +0.000028] usb 5-1.2: Detected FT4232H
[  +0.000614] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB1
[  +0.000356] ftdi_sio 5-1.2:1.2: FTDI USB Serial Device converter detected
[  +0.000036] usb 5-1.2: Detected FT4232H
[  +0.000620] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB2
[  +0.000356] ftdi_sio 5-1.2:1.3: FTDI USB Serial Device converter detected
[  +0.000032] usb 5-1.2: Detected FT4232H
[  +0.000600] usb 5-1.2: FTDI USB Serial Device converter now attached to ttyUSB3
[  +0.075083] usb 5-1.3: new full-speed USB device number 56 using xhci_hcd
[  +0.108450] usb 5-1.3: New USB device found, idVendor=18d1, idProduct=520e, bcdDevice= 1.00
[  +0.000011] usb 5-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  +0.000004] usb 5-1.3: Product: HyperDebug CMSIS-DAP
[  +0.000004] usb 5-1.3: Manufacturer: Google LLC
[  +0.000004] usb 5-1.3: SerialNumber: 206D37965853
[  +0.007996] usb_serial_simple 5-1.3:1.0: google converter detected
[  +0.000171] usb 5-1.3: google converter now attached to ttyUSB4
[  +0.008794] usb_serial_simple 5-1.3:1.3: google converter detected
[  +0.000111] usb 5-1.3: google converter now attached to ttyUSB5
[  +0.002871] usb_serial_simple 5-1.3:1.4: google converter detected
[  +0.000066] usb 5-1.3: google converter now attached to ttyUSB6
[  +0.002931] usb_serial_simple 5-1.3:1.5: google converter detected
[  +0.000062] usb 5-1.3: google converter now attached to ttyUSB7
[  +0.002967] usb_serial_simple 5-1.3:1.6: google converter detected
[  +0.000078] usb 5-1.3: google converter now attached to ttyUSB8
[  +0.180485] usb 5-1.4: new full-speed USB device number 57 using xhci_hcd
[  +0.073063] usb 5-1.4: device descriptor read/64, error -32
[  +0.181124] usb 5-1.4: device descriptor read/64, error -32
[  +0.184177] usb 5-1.4: new full-speed USB device number 58 using xhci_hcd
[  +0.075778] usb 5-1.4: device descriptor read/64, error -32
[  +0.178911] usb 5-1.4: device descriptor read/64, error -32
[  +0.028392] usb 5-1-port4: attempt power cycle
[  +0.587189] usb 5-1.4: new full-speed USB device number 59 using xhci_hcd
[  +0.000263] usb 5-1.4: Device not responding to setup address.
[  +0.204934] usb 5-1.4: Device not responding to setup address.
[  +0.207888] usb 5-1.4: device not accepting address 59, error -71
[  +0.000518] usb 5-1.4: WARN: invalid context state for evaluate context command.
[  +0.074400] usb 5-1.4: new full-speed USB device number 60 using xhci_hcd
[  +0.000262] usb 5-1.4: Device not responding to setup address.
[  +0.204910] usb 5-1.4: Device not responding to setup address.
[  +0.207842] usb 5-1.4: device not accepting address 60, error -71
[  +0.000437] usb 5-1.4: WARN: invalid context state for evaluate context command.
[  +0.000345] usb 5-1-port4: unable to enumerate USB device
[  +0.181255] usb 5-1.1: new high-speed USB device number 61 using xhci_hcd
[  +0.095274] usb 5-1.1: New USB device found, idVendor=2b3e, idProduct=c340, bcdDevice= 9.00
[  +0.000016] usb 5-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  +0.000007] usb 5-1.1: Product: ChipWhisperer CW340 - Luna Board
[  +0.000006] usb 5-1.1: Manufacturer: NewAE Technology Inc.
[  +0.000005] usb 5-1.1: SerialNumber: 51203220353053523230382037313038
[  +0.070690] cdc_acm 5-1.1:1.1: ttyACM0: USB ACM device
[  +0.001055] cdc_acm 5-1.1:1.3: ttyACM1: USB ACM device
```

> 7. Depending on your case, do one of the following:

Reviewer: ```suggestion
7. Depending on your use case, do one of the following:
```
or maybe "setup"?

   1. If you are connecting a HyperDebug board to your CW340 base board, follow instructions in the [HyperDebug Board](#hyperdebug-board) section.
   2. Otherwise, move the *Control Power* switch (top left corner, *SW7*) to the left (towards the barrel jack) to power on the board.

> Run the following tests to confirm that the dip switches are set up correctly:

Reviewer: You still need to do more setup for this with HyperDebug before you can run these. I would clarify this:

```suggestion
After completing the rest of the FPGA setup process, you can confirm that these DIP switches are configured correctly by running the following test targets:
```

### Detecting the PC Connections to the Board(s)

> To detect if you PC has successfully connected to your FPGA and/or HyperDebug boards, you can use the following command to monitor output from dmesg:

Reviewer: Great fix, but this error actually happens twice :)
```suggestion
To detect if your PC has successfully connected to your FPGA and/or HyperDebug boards, you can use the following command to monitor output from dmesg:
```

Author: Haha, didn't notice it.
